api/lib/reducers/polkadotV0-reducers.js

```diff
@@ -470,6 +491,162 @@ function rewardReducer(network, validators, reward, reducers) {
+    deposit: toViewDenom(network, Number(proposal.proposal.bond)),
+    proposer: proposal.proposal.proposer.toHuman(),
+    beneficiary: proposal.proposal.beneficiary // the account getting the tip
+  }
+}
+
+function councilProposalReducer(
+  network,
+  proposal,
+  councilMembers,
+  blockHeight
+) {
+  return {
+    id: proposal.votes.index,
+    networkId: network.id,
+    type: proposalTypeEnum.COUNCIL,
+    title: `Council Proposal #${proposal.votes.index}`,
+    description: proposal.description,
+    creationTime: proposal.creationTime,
+    status: `VotingPeriod`,
+    statusBeginTime: proposal.creationTime,
+    statusEndTime: getStatusEndTime(blockHeight, proposal.votes.end),
+    tally: councilTallyReducer(proposal.votes, councilMembers),
+    deposit: undefined,
+    proposer: undefined
+  }
+}
+
+function tallyReducer(network, tally, totalIssuance) {
+  //
+  // tally chain format:
+  //
+  // "tally": {
+  //   "ayes": "0x0000000000000000001e470441298100",
+  //   "nays": "0x00000000000000000186de726fc56000",
+  //   "turnout": "0x000000000000000000dc3dd9ad3d0800"
+  // }
+  //
+  // turnout is the real amount deposited by voters
+  // in polkadot you can vote with "conviction", that means
+  // ayes and nays are amplified by the selected lockup period:
+  //
+  // 1x voting balance, locked for 1x enactment (8.00 days)
+  // 2x voting balance, locked for 2x enactment (16.00 days)
+  // 3x voting balance, locked for 4x enactment (32.00 days)
+  // 4x voting balance, locked for 8x enactment (64.00 days)
+  // 5x voting balance, locked for 16x enactment (128.00 days)
+  // 6x voting balance, locked for 32x enactment (256.00 days)
+  //
+
+  const turnout = BigNumber(tally.turnout)
+
+  const totalVoted = BigNumber(tally.ayes).plus(tally.nays)
+  const total = toViewDenom(network, totalVoted.toString(10))
+  const yes = toViewDenom(network, tally.ayes)
+  const no = toViewDenom(network, tally.nays)
+  const totalVotedPercentage = turnout
+    .div(BigNumber(totalIssuance))
+    .toNumber()
+    .toFixed(4) // the percent conversion is done in the FE. No need to multiply by 100
+
+  return {
+    yes,
+    no,
+    abstain: 0,
+    veto: 0,
+    total,
+    totalVotedPercentage
+  }
+}
+
+function councilTallyReducer(votes, councilMembers) {
+  const total = votes.ayes.length + votes.nays.length
+  return {
+    yes: votes.ayes.length,
+    no: votes.nays.length,
+    abstain: 0,
+    veto: 0,
+    total,
+    totalVotedPercentage: (total / councilMembers.length).toFixed(4) // the percent conversion is done in the FE. No need to multiply by 100
+  }
+}
+
+function democracyTallyReducer(proposal) {
+  // if we consider democracyProposals like the parallel to Cosmos proposals in deposit periods, then
+  // we would have to turn the seconds concept into a deposit somehow
+  return {
+    yes: proposal.seconds.length
+  }
+}
+
+function toViewDenom(network, chainDenomAmount) {
```

Comment:
this is douplicated. can we refactor this function into a common lib or put it into the network object? (we need to have a network getter function because enriching the networks object currently needs to be done in a variety of places.)

Reply:
where do we need to enrich the network object concretely? You lost me there

Reply:
I personally think it is fine to have `toViewDenom` as a common function in numbers.js

Reply:
let's move it to numbers.js for now. I open an issue about the refactoring

Reply:
https://github.com/luniehq/lunie/issues/4664